contracts/voting/src/lib.rs

```diff
@@ -209,14 +318,21 @@ impl ProposalVotingContract {
 #[derive(Clone, Debug, PartialEq)]
 pub struct Proposal {
     id: u64,
+    // Allows external systems to discriminate among type of proposal. This probably
+    // goes in hand with the `comment` field.
+    kind: ProposalType,
+    // The address this proposal is created from.
+    proposer: Address,
+    // Comment has enough size for a wasm contract hash. It could also be a string.
```

Comment:
Why do we need a contract hash?

Reply:
We need a contract hash in order to execute the code upgrade [later](https://github.com/eigerco/nebula/pull/51/files#diff-7b82d5fb59b61725f07142c97d6337836600e3918432726ceb89bc7f73b88cb8R335). Depending of the `kind` of the proposal, the `comment` field is decoded one way or the another. Thats the idea.